packages/botonic-core/src/debug/inspector.ts

```diff
@@ -53,30 +74,29 @@ export class LogRouteInspector extends FocusRouteInspector {
     )
   }
 
-  _routeName(obj) {
+  _routeName(obj: Route): string {
     let name = obj.path || '<<no path defined>>'
     if (obj.action) {
       name = `'${name}' (to action '${obj.action.name}')`
     }
     if (obj.redirect) {
       name = `'${name}' (with redirect to '${obj.redirect}')`
     }
-
-    return name
+    return name as string
```

Comment:
do you need to do this due to `obj.path`? All the rest are strings, so what about doing so ASAP, `obj.path as string`?

Reply:
I see.. this comes from the argument typed as Route, which has typed path as a stringMatcher, that can be a regex:
```
export interface Route {
  ...
  path?: StringMatcher --> `RegExp | string | ((data: string) => boolean)`
```
hence the "trick". The better solution here pass by strictly reviewing all `@botonic/core` and also try to split complex functions like `processInput` into smaller pieces and with unit tests. I would leave it as is atm and reviewing eventually when working with this package.